base/reduce.jl

```diff
@@ -954,30 +953,30 @@ julia> count([true, false, true, true])
 3
 ```
 """
-count(itr) = count(identity, itr)
+count(itr; init=0) = count(identity, itr; init)
 
-count(f, itr) = _simple_count(f, itr)
+count(f, itr; init=0) = _simple_count(f, itr, init)
 
-function _simple_count(pred, itr)
-    n = 0
+function _simple_count(pred, itr, init)
+    n = init
     for x in itr
         n += pred(x)::Bool
     end
     return n
 end
 
-function count(::typeof(identity), x::Array{Bool})
-    n = 0
+function _simple_count(::typeof(identity), x::Array{Bool}, init::T=0) where {T}
```

Comment:
This is actually a bugfix (although it doesn't occur in 1.5), because `count(::Array{Bool})` currently errors when passed a `dims` argument.